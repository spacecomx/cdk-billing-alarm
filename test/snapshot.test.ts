import '@aws-cdk/assert/jest';
import { SynthUtils } from '@aws-cdk/assert';
import {
  BasicBillingAlarmTest,
  BillingAlarmWithoutEmailSubscriptionTest,
  BillingAlarmWithCustomResourceNamesTest,
  LinkExistingTopicArnToBillingAlarmTest,
  LinkBillingAlarmToAwsAccountIdTest,
  LinkBillingAlarmToAwsServiceTest,
  LinkBillingAlarmToAwsAccountIdToAwsServiceTest,
} from '../src/integ.default';

test('integration snapshot test for basic billing alarm configuation', ()=> {
  const integration = new BasicBillingAlarmTest();

  integration.stack.forEach(stack => {
    expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
  });
});

test('integration snapshot test for billing alarm without email address subscription to sns topic', ()=> {
  const integration = new BillingAlarmWithoutEmailSubscriptionTest();

  integration.stack.forEach(stack => {
    expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
  });
});

test('integration snapshot test for billing alarm custom resource naming', ()=> {
  const integration = new BillingAlarmWithCustomResourceNamesTest();

  integration.stack.forEach(stack => {
    expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
  });
});

test('integration snapshot test for linking existing sns topic arn to billing alarm', ()=> {
  const integration = new LinkExistingTopicArnToBillingAlarmTest();

  integration.stack.forEach(stack => {
    expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
  });
});

test('integration snapshot test for linking an aws account id to billing alarm', ()=> {
  const integration = new LinkBillingAlarmToAwsAccountIdTest();

  integration.stack.forEach(stack => {
    expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
  });
});

test('integration snapshot test for linking an aws service to billing alarm', ()=> {
  const integration = new LinkBillingAlarmToAwsServiceTest();

  integration.stack.forEach(stack => {
    expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
  });
});

test('integration snapshot test for linking aws account id and service to billing alarm', ()=> {
  const integration = new LinkBillingAlarmToAwsAccountIdToAwsServiceTest();

  integration.stack.forEach(stack => {
    expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
  });
});