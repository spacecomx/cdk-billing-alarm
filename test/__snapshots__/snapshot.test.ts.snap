// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`integration snapshot test for basic billing alarm configuation 1`] = `
Object {
  "Outputs": Object {
    "BillingAlarmTopicArn5F4EC7DB": Object {
      "Description": "SNS Topic Arn.",
      "Value": Object {
        "Ref": "BillingAlarmTopic524A3288",
      },
    },
  },
  "Resources": Object {
    "BillingAlarm990F724D": Object {
      "Properties": Object {
        "AlarmActions": Array [
          Object {
            "Ref": "BillingAlarmTopic524A3288",
          },
        ],
        "ComparisonOperator": "GreaterThanOrEqualToThreshold",
        "Dimensions": Array [
          Object {
            "Name": "Currency",
            "Value": "USD",
          },
        ],
        "EvaluationPeriods": 1,
        "MetricName": "EstimatedCharges",
        "Namespace": "AWS/Billing",
        "Period": 21600,
        "Statistic": "Maximum",
        "Threshold": 20,
      },
      "Type": "AWS::CloudWatch::Alarm",
    },
    "BillingAlarmTopic524A3288": Object {
      "Type": "AWS::SNS::Topic",
    },
    "BillingAlarmTopicjohnexampleorg85872015": Object {
      "Properties": Object {
        "Endpoint": "john@example.org",
        "Protocol": "email",
        "TopicArn": Object {
          "Ref": "BillingAlarmTopic524A3288",
        },
      },
      "Type": "AWS::SNS::Subscription",
    },
  },
}
`;

exports[`integration snapshot test for billing alarm custom resource naming 1`] = `
Object {
  "Outputs": Object {
    "BillingAlarmTopicArn5F4EC7DB": Object {
      "Description": "SNS Topic Arn.",
      "Value": Object {
        "Ref": "BillingAlarmTopic524A3288",
      },
    },
  },
  "Resources": Object {
    "BillingAlarm990F724D": Object {
      "Properties": Object {
        "AlarmActions": Array [
          Object {
            "Ref": "BillingAlarmTopic524A3288",
          },
        ],
        "AlarmDescription": "Alarm threshold alert for every AWS Service used in the account",
        "AlarmName": "Billing Alarm (All Services)",
        "ComparisonOperator": "GreaterThanOrEqualToThreshold",
        "Dimensions": Array [
          Object {
            "Name": "Currency",
            "Value": "USD",
          },
        ],
        "EvaluationPeriods": 1,
        "MetricName": "EstimatedCharges",
        "Namespace": "AWS/Billing",
        "Period": 21600,
        "Statistic": "Maximum",
        "Threshold": 20,
      },
      "Type": "AWS::CloudWatch::Alarm",
    },
    "BillingAlarmTopic524A3288": Object {
      "Properties": Object {
        "DisplayName": "Billing alarm topic for monthly alarm estimates",
        "TopicName": "BillingAlarmTopic",
      },
      "Type": "AWS::SNS::Topic",
    },
    "BillingAlarmTopicjohnexampleorg85872015": Object {
      "Properties": Object {
        "Endpoint": "john@example.org",
        "Protocol": "email",
        "TopicArn": Object {
          "Ref": "BillingAlarmTopic524A3288",
        },
      },
      "Type": "AWS::SNS::Subscription",
    },
  },
}
`;

exports[`integration snapshot test for billing alarm without email address subscription to sns topic 1`] = `
Object {
  "Outputs": Object {
    "BillingAlarmTopicArn5F4EC7DB": Object {
      "Description": "SNS Topic Arn.",
      "Value": Object {
        "Ref": "BillingAlarmTopic524A3288",
      },
    },
  },
  "Resources": Object {
    "BillingAlarm990F724D": Object {
      "Properties": Object {
        "AlarmActions": Array [
          Object {
            "Ref": "BillingAlarmTopic524A3288",
          },
        ],
        "ComparisonOperator": "GreaterThanOrEqualToThreshold",
        "Dimensions": Array [
          Object {
            "Name": "Currency",
            "Value": "USD",
          },
        ],
        "EvaluationPeriods": 1,
        "MetricName": "EstimatedCharges",
        "Namespace": "AWS/Billing",
        "Period": 21600,
        "Statistic": "Maximum",
        "Threshold": 20,
      },
      "Type": "AWS::CloudWatch::Alarm",
    },
    "BillingAlarmTopic524A3288": Object {
      "Type": "AWS::SNS::Topic",
    },
  },
}
`;

exports[`integration snapshot test for linking an aws account id to billing alarm 1`] = `
Object {
  "Outputs": Object {
    "BillingAlarmTopicArn5F4EC7DB": Object {
      "Description": "SNS Topic Arn.",
      "Value": Object {
        "Ref": "BillingAlarmTopic524A3288",
      },
    },
  },
  "Resources": Object {
    "BillingAlarm990F724D": Object {
      "Properties": Object {
        "AlarmActions": Array [
          Object {
            "Ref": "BillingAlarmTopic524A3288",
          },
        ],
        "AlarmDescription": "Billing Alarm all services (Account: 123456789000)",
        "AlarmName": "Billing Alarm All Services (Account: 123456789000)",
        "ComparisonOperator": "GreaterThanOrEqualToThreshold",
        "Dimensions": Array [
          Object {
            "Name": "Currency",
            "Value": "USD",
          },
          Object {
            "Name": "LinkedAccount",
            "Value": "123456789000",
          },
        ],
        "EvaluationPeriods": 1,
        "MetricName": "EstimatedCharges",
        "Namespace": "AWS/Billing",
        "Period": 21600,
        "Statistic": "Maximum",
        "Threshold": 20,
      },
      "Type": "AWS::CloudWatch::Alarm",
    },
    "BillingAlarmTopic524A3288": Object {
      "Properties": Object {
        "DisplayName": "Billing alarm topic for monthly alarm estimates",
      },
      "Type": "AWS::SNS::Topic",
    },
    "BillingAlarmTopicjohnexampleorg85872015": Object {
      "Properties": Object {
        "Endpoint": "john@example.org",
        "Protocol": "email",
        "TopicArn": Object {
          "Ref": "BillingAlarmTopic524A3288",
        },
      },
      "Type": "AWS::SNS::Subscription",
    },
  },
}
`;

exports[`integration snapshot test for linking an aws service to billing alarm 1`] = `
Object {
  "Outputs": Object {
    "BillingAlarmTopicArn5F4EC7DB": Object {
      "Description": "SNS Topic Arn.",
      "Value": Object {
        "Ref": "BillingAlarmTopic524A3288",
      },
    },
  },
  "Resources": Object {
    "BillingAlarm990F724D": Object {
      "Properties": Object {
        "AlarmActions": Array [
          Object {
            "Ref": "BillingAlarmTopic524A3288",
          },
        ],
        "AlarmDescription": "Billing Alarm for all Amazon S3 charges",
        "AlarmName": "Billing Alarm (Amazon S3)",
        "ComparisonOperator": "GreaterThanOrEqualToThreshold",
        "Dimensions": Array [
          Object {
            "Name": "Currency",
            "Value": "USD",
          },
          Object {
            "Name": "ServiceName",
            "Value": "AmazonS3",
          },
        ],
        "EvaluationPeriods": 1,
        "MetricName": "EstimatedCharges",
        "Namespace": "AWS/Billing",
        "Period": 21600,
        "Statistic": "Maximum",
        "Threshold": 20,
      },
      "Type": "AWS::CloudWatch::Alarm",
    },
    "BillingAlarmTopic524A3288": Object {
      "Properties": Object {
        "DisplayName": "Billing alarm topic for monthly alarm estimates",
      },
      "Type": "AWS::SNS::Topic",
    },
    "BillingAlarmTopicjohnexampleorg85872015": Object {
      "Properties": Object {
        "Endpoint": "john@example.org",
        "Protocol": "email",
        "TopicArn": Object {
          "Ref": "BillingAlarmTopic524A3288",
        },
      },
      "Type": "AWS::SNS::Subscription",
    },
  },
}
`;

exports[`integration snapshot test for linking aws account id and service to billing alarm 1`] = `
Object {
  "Outputs": Object {
    "BillingAlarmTopicArn5F4EC7DB": Object {
      "Description": "SNS Topic Arn.",
      "Value": Object {
        "Ref": "BillingAlarmTopic524A3288",
      },
    },
  },
  "Resources": Object {
    "BillingAlarm990F724D": Object {
      "Properties": Object {
        "AlarmActions": Array [
          Object {
            "Ref": "BillingAlarmTopic524A3288",
          },
        ],
        "AlarmDescription": "Billing Alarm for all Amazon S3 charges (Account: 123456789000)",
        "AlarmName": "Billing Alarm Amazon S3 (Account: 123456789000)",
        "ComparisonOperator": "GreaterThanOrEqualToThreshold",
        "Dimensions": Array [
          Object {
            "Name": "Currency",
            "Value": "USD",
          },
          Object {
            "Name": "LinkedAccount",
            "Value": "123456789000",
          },
          Object {
            "Name": "ServiceName",
            "Value": "AmazonS3",
          },
        ],
        "EvaluationPeriods": 1,
        "MetricName": "EstimatedCharges",
        "Namespace": "AWS/Billing",
        "Period": 21600,
        "Statistic": "Maximum",
        "Threshold": 20,
      },
      "Type": "AWS::CloudWatch::Alarm",
    },
    "BillingAlarmTopic524A3288": Object {
      "Properties": Object {
        "DisplayName": "Billing alarm topic for monthly alarm estimates",
      },
      "Type": "AWS::SNS::Topic",
    },
    "BillingAlarmTopicjohnexampleorg85872015": Object {
      "Properties": Object {
        "Endpoint": "john@example.org",
        "Protocol": "email",
        "TopicArn": Object {
          "Ref": "BillingAlarmTopic524A3288",
        },
      },
      "Type": "AWS::SNS::Subscription",
    },
  },
}
`;

exports[`integration snapshot test for linking existing sns topic arn to billing alarm 1`] = `
Object {
  "Outputs": Object {
    "BillingAlarmTopicArn5F4EC7DB": Object {
      "Description": "SNS Topic Arn.",
      "Value": "arn:aws:sns:us-east-1:444455556666:BillingAlarmTopic",
    },
  },
  "Resources": Object {
    "BillingAlarm990F724D": Object {
      "Properties": Object {
        "AlarmActions": Array [
          "arn:aws:sns:us-east-1:444455556666:BillingAlarmTopic",
        ],
        "ComparisonOperator": "GreaterThanOrEqualToThreshold",
        "Dimensions": Array [
          Object {
            "Name": "Currency",
            "Value": "USD",
          },
        ],
        "EvaluationPeriods": 1,
        "MetricName": "EstimatedCharges",
        "Namespace": "AWS/Billing",
        "Period": 21600,
        "Statistic": "Maximum",
        "Threshold": 20,
      },
      "Type": "AWS::CloudWatch::Alarm",
    },
    "BillingAlarmTopicjohnexampleorg85872015": Object {
      "Properties": Object {
        "Endpoint": "john@example.org",
        "Protocol": "email",
        "TopicArn": "arn:aws:sns:us-east-1:444455556666:BillingAlarmTopic",
      },
      "Type": "AWS::SNS::Subscription",
    },
  },
}
`;
